### PR TITLE
POST/GET for gamer activity

### PR DESCRIPTION
**Pull request by vickiszhang**
relates to #23 

- GET api to get gamer acitivity for a specific student number
- POST api to insert a new gamer activity row
- PATCH api to update row with end time given student number

todo:
- delete
- unspaghetti code

notes:
- i also changed the started_at/ended_at column types to timestamp, since I assume we would want to know the specific time someone is in the lounge

### CODE REVIEW

**Review by joshhwuu — COMMENTED** [review]
Should get some tests going too, feel free to make a new file in `tests/` or just add a TODO for now

File: client/app/lounge/activity.tsx [hunk 1]
```diff
@@ -1,7 +1,207 @@
+import { useState } from "react";
```
Comment (Jaden51) [inline client/app/lounge/activity.tsx:1]: The requirements for this issue were only the backend API. Please remove this file 😄 

The frontend form for this API is being made here if you are curious 😃 https://github.com/ubcesports/lounge-hub/pull/28. 

This was helpful tho to look at how the API was called on the frontend so I'm going to take note of this commit hash and perhaps use some of the code when I get to it on my PR above 😄

File: app/app.js [hunk 2]
```diff
@@ -142,6 +142,119 @@ app.delete("/api/gamer/:student_number", async (req, res) => {
...
+/**
+ * @api {get} /activity/:student_number Get Gamer Activity
+ * @apiName GetGamerActivity
+ * @apiGroup Activity
+ *
+ * @apiParam {String} student_number Student's unique number.
+ *
+ * @apiSuccess {Object} gamer_activity Gamer activity object.
+ * @apiSuccess {String} gamer_activity.student_number Student number, 8 digit integer.
+ * @apiSuccess {String} gamer_activity.pc_number PC number.
```
Comment (Jaden51) [inline app/app.js:154]: PC number should return an integer

**Review by Jaden51 — COMMENTED** [review]
Awesome work!! I love how you managed to update the ended_at column! One thing, could you add an API call for fetching the last 20 or something rows in the table? The amount I'll leave to you but we need an API to see all recent activity 😄

File: app/app.js [hunk 3]
```diff
@@ -142,6 +142,154 @@ app.delete("/api/gamer/:student_number", async (req, res) => {
...
+ * @apiSuccess {String} gamer_activity.started_at Datetime when the activity started.
+ * @apiSuccess {String} gamer_activity.ended_at Datetime when the activity ended.
+ *
+ * @apiError {String} 500 Server error.
+ */
+app.get("/api/activity/all/recent", async (req, res) => {
+  try {
+    const query = `SELECT * FROM users_test.gamer_activity
+      ORDER BY started_at 
+      DESC LIMIT 20;`;
```
Comment (Jaden51) [inline app/app.js:193]: PostgreSQL by default sorts NULL values to be higher than any other value. This means what when we order by DESC all the null values go to the top. Could you order by `NULLS LAST` 

Something like
```    
const query = `SELECT * FROM users_test.gamer_activity
      ORDER BY started_at 
      DESC NULLS LAST
      LIMIT 20;`;
```

**Comment by Jaden51** [comment]
After you resolve my comment above, everything else looks good!

**Review by joshhwuu — APPROVED** [review]
This LGTM, good work! 

Something else I noticed: since our API have db path hard coded in, creating a "test table" in our tests actually doesn't prevent running tests from creating new rows in the actual table as well. My fault on the oversight, I'll add in new schemas for testing and "actual" data

**Pull request merged**